### PR TITLE
Edit Content: Add JSON Field

### LINKED ISSUE #26792: Edit Content: Add JSON Field
### Parent Issue

https://github.com/dotCMS/core/issues/25445

### Task

As a user, I want to have access to the JSON field in the new `Edit Content`.

### Proposed Objective

Code Maintenance

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

The new JSON Field should allow the user to create JSON Objects.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

We need to configure Monaco Editor to set the highlight for JSON syntax.

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by rjvelazco**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 33f1560</samp>

Added a new component `dot-edit-content-json-field` to handle JSON fields in the content editor. This component uses the ngx-monaco-editor library to provide a code editor with JSON syntax highlighting and validation. Updated the `dot-edit-content-fields` module and the `dot-edit-content-field` component to support the new field type.

## Related Issue
Fixes #26792

# Explanation of Changes

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 33f1560</samp>

*  Add support for JSON field type in content editor ([link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-685da2f87189fff825b7aec2666cfbabdb33c40f8f9d8c82df77ebb8652ccd4aR51-R55), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-4c8a279470ceb366e4f859ef3a36b8678702237576f7e4d9db1622fd9b0f8978R6), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-4c8a279470ceb366e4f859ef3a36b8678702237576f7e4d9db1622fd9b0f8978L24-R26), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-4c8a279470ceb366e4f859ef3a36b8678702237576f7e4d9db1622fd9b0f8978L35-R38), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-b8aa90bf87b74a086b7d91cd7a45b06022453dba3f26a7fd751a7fd2aacb2f42R1-R4), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-4735fe6e21881c40efde1ce93f58c4ec3a677e0a58d66578159b79828532d04dR1-R11), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-7de4eab63be6786869cc64142ef14ec11998fe5d7aa9bfd3c9a38979ebbd5216R1-R22), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-cb3c5e92c52c399050b6321834f69fe6125ce05b964f3fdd5425d7ae0e623026R1-R38), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-aa72329fa1c18891b5d44042ac6ae09ff3d915621f16a084982f87d068c2453aL23-R24))
  * Import and declare `dot-edit-content-json-field` component in `dot-edit-content-fields` module ([link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-4c8a279470ceb366e4f859ef3a36b8678702237576f7e4d9db1622fd9b0f8978R6), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-4c8a279470ceb366e4f859ef3a36b8678702237576f7e4d9db1622fd9b0f8978L24-R26), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-4c8a279470ceb366e4f859ef3a36b8678702237576f7e4d9db1622fd9b0f8978L35-R38))
  * Render `dot-edit-content-json-field` component in `dot-edit-content-field` component template when field type is JSON ([link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-685da2f87189fff825b7aec2666cfbabdb33c40f8f9d8c82df77ebb8652ccd4aR51-R55))
  * Define `dot-edit-content-json-field` component template, style sheet, and logic ([link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-b8aa90bf87b74a086b7d91cd7a45b06022453dba3f26a7fd751a7fd2aacb2f42R1-R4), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-4735fe6e21881c40efde1ce93f58c4ec3a677e0a58d66578159b79828532d04dR1-R11), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-cb3c5e92c52c399050b6321834f69fe6125ce05b964f3fdd5425d7ae0e623026R1-R38))
    * Use `ngx-monaco-editor` component to display a code editor with JSON syntax highlighting and validation ([link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-b8aa90bf87b74a086b7d91cd7a45b06022453dba3f26a7fd751a7fd2aacb2f42R1-R4), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-cb3c5e92c52c399050b6321834f69fe6125ce05b964f3fdd5425d7ae0e623026R1-R38))
    * Bind form control name to field variable and pass editor options to editor component ([link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-b8aa90bf87b74a086b7d91cd7a45b06022453dba3f26a7fd751a7fd2aacb2f42R1-R4), [link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-cb3c5e92c52c399050b6321834f69fe6125ce05b964f3fdd5425d7ae0e623026R1-R38))
    * Use `ChangeDetectionStrategy.OnPush` and `viewProviders` to improve performance and inject parent form group ([link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-cb3c5e92c52c399050b6321834f69fe6125ce05b964f3fdd5425d7ae0e623026R1-R38))
    * Declare input property for field and constant for editor options ([link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-cb3c5e92c52c399050b6321834f69fe6125ce05b964f3fdd5425d7ae0e623026R1-R38))
  * Add JSON field type to `dot-edit-content-field` enum ([link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-aa72329fa1c18891b5d44042ac6ae09ff3d915621f16a084982f87d068c2453aL23-R24))
  * Test `dot-edit-content-json-field` component creation ([link](https://github.com/dotCMS/core/pull/26797/files?diff=unified&w=0#diff-7de4eab63be6786869cc64142ef14ec11998fe5d7aa9bfd3c9a38979ebbd5216R1-R22))
  
  
  #### ScreenShot
 
JSON Field in the Edit Content

#### Video

https://github.com/dotCMS/core/assets/72418962/5cc20f79-afa2-4d07-8657-e9d9a9452a0c

### CODE REVIEW

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div class="json-field__container" data-testId="json-field-container" style="resize: 'vertical'">
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.html:1]: Is this inline style necessary?

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.html (hunk 1 above)
Comment (fmontes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.html:1]: We can do this in the `:host` for sure :)

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    ngOnInit(): void {
+        const form = this.controlContainer.control;
+        const control = form.get(this.field.variable);
+
+        /*
+         * This is a workaround to force the change detection to run when the value of the control changes.
+         * This is needed because the Monaco Editor does not play well with the change detection strategy of the component.
+         */
+        control.valueChanges.subscribe(() => this.cd.markForCheck());
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.ts:57]: Does this not need a destroy?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
+@use "variables" as *;
+
+.json-field__container {
+    min-height: 9.375rem; // 150px
+    height: 100%;
+    width: 100%;
+    overflow: auto;
+}
+
+.json-field__editor {
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.scss:10]:
```suggestion
ngx-monaco-editor {
```

Is only one... just saying :)

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.html (hunk 1 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.html:1]: I added that inline style to test it

Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.html:1]: Done!

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.scss (hunk 3 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.scss:10]: Done!

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.ts (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-json-field/dot-edit-content-json-field.component.ts:57]: Done!

**Review by oidacra — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**